compiler/packages/snap/src/runner-watch.ts

```diff
@@ -189,14 +189,15 @@ function subscribeKeyEvents(
   state: RunnerState,
   onChange: (state: RunnerState) => void
 ) {
-  process.stdin.on("keypress", (str, key) => {
+  process.stdin.on("keypress", async (str, key) => {
     if (key.name === "u") {
       // u => update fixtures
       state.mode.action = RunnerAction.Update;
     } else if (key.name === "q") {
       process.exit(0);
     } else if (key.name === "f") {
       state.mode.filter = !state.mode.filter;
+      state.filter = state.mode.filter ? await readTestFilter() : null;
```

Comment:
having an async event handler is a bit weird, but technically this is already async since `onChange` can be async, it just isn't expressed in the type signature. meh, probably fine to land this.